tests/phpunit/tests/functions/getThePostsNavigation.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * Tests get_the_posts_navigation function.
+ *
+ * @ticket 55751
+ * @since 6.1.0
+ *
+ * @covers ::get_the_posts_navigation
+ */
+class Tests_Functions_GetThePostsNavigation extends WP_UnitTestCase {
+
+	/**
+	 * Create posts for the tests.
+	 *
+	 * @param WP_UnitTest_Factory $factory
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		$factory->post->create_many( 3 );
+	}
+
+	/**
+	 * Run tests on `get_the_posts_navigation()`.
+	 *
+	 * @dataProvider data_get_the_posts_navigation
+	 *
+	 * @param int    $post_per_page      Posts per page to be queried.
+	 * @param int    $paged_num          Pagination page number.
+	 * @param bool   $assert_older_posts Assert older posts nav string.
+	 * @param bool   $assert_newer_posts Assert newer posts nav string.
+	 * @param bool   $assert_empty       Assert empty posts nav string.
+	 */
+	public function test_get_the_posts_navigation(
+		$post_per_page, $paged_num,
+		$assert_older_posts,
+		$assert_newer_posts,
+		$assert_empty ) {
```

Comment:
```suggestion
	public function test_get_the_posts_navigation( $posts_per_page, $paged_num, $expected, $message ) {
```
By removing the `$assert_` arguments and replacing with `$expected` and `$message`, you can simply use a single assertion with no conditionals, a shorter docblock and no multiline method signature:

```php
$this->assertSame( $expected, get_the_posts_navigation(), $message );
```

Reply:
Good recommendation. Thanks.

Reply:
@costdev, the only problem I see is that there is a case where `get_the_posts_navigation()` returns both `Older posts` and `Newer posts` as strings and I cannot passed that as `$expected`  to be checked with `assertStringContainsString`. I think that was the reason why I opted with the true/false approach.

Reply:
Maybe an option could be checking the actual HTML output with `assertRegExp`?

Reply:
Hmm, good point. Sorry for missing that!

In that case, I think the following changes will simplify this test's docblock, signature and data provider:

- Use shorter argument names.
- Remove `$assert_empty`, as this is effectively when `$older` and `$newer` are both `false`, and is therefore redundant.

The outcome would be:
```php
/**
 * Run tests on `get_the_posts_navigation()`.
 *
 * @dataProvider data_get_the_posts_navigation
 *
 * @param int  $per_page  Posts per page to be queried.
 * @param int  $paged_num Pagination page number.
 * @param bool $older     Whether an "Older posts" link should be included.
 * @param bool $newer     Whether a "Newer posts" link should be included.
 */
public function test_get_the_posts_navigation( $per_page, $paged_num, $older, $newer ) {
	global $wp_query, $paged;

	$paged    = $paged_num;
	$wp_query = new WP_Query(
		array(
			'post_type'      => 'post',
			'posts_per_page' => $per_page,
			'paged'          => $paged,
		)
	);

	$actual = get_the_posts_navigation();

	if ( $older ) {
		$this->assertStringContainsString(
			'Older posts',
			$actual,
			'Posts navigation must contain an "Older posts" link.'
		);
	}

	if ( $newer ) {
		$this->assertStringContainsString(
			'Newer posts',
			$actual,
			'Posts navigation must contain a "Newer posts" link.'
		);
	}

	if ( ! $older && ! $newer ) {
		$this->assertEmpty(
			$actual,
			'Posts navigation must be an empty string.'
		);
	}
}
```